### PR TITLE
[CWS] remove watch dir option from directory provider

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

With remote config this option has lost its utility. It was also disabled by default (and has been for ever), and can be replicate with a SIGHUP on system-probe if needed.

Removing it allows to clean up this code, but also to remove the fsnotify dependency on pkg/security/secl.

### CODE REVIEW

**Review by spikat — APPROVED** [review]

**Review by YoannGh — APPROVED** [review]

**Review by usamasaqib — APPROVED** [review]

File: releasenotes/notes/cws-remove-watch-dir-option-f8faee4937353316.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+deprecations:
+  - |
+    CWS: the `runtime_security_config.policies.watch_dir` option has been removed.
+    Please use remote configuration for dynamically updating policies, or simply send
```
Comment (neko-dd) [inline releasenotes/notes/cws-remove-watch-dir-option-f8faee4937353316.yaml:12]:
```suggestion
    Use remote configuration for dynamically updating policies, or send
```

File: releasenotes/notes/cws-remove-watch-dir-option-f8faee4937353316.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    the SIGHUP signal to the system-probe process to reload the policies.
```
Comment (neko-dd) [inline releasenotes/notes/cws-remove-watch-dir-option-f8faee4937353316.yaml:13]:
```suggestion
    the `SIGHUP` signal to the `system-probe` process to reload the policies.
```

Assuming these are both code literals

**Review by neko-dd — APPROVED** [review]

**Pull request merged**